eth_abi/registry.py

```diff
@@ -434,6 +438,20 @@ def unregister(self, label: str) -> None:
     def get_encoder(self, type_str):
         return self._get_coder(self._encoders, type_str)
 
+    def has_encoder(self, type_str: abi.TypeStr) -> bool:
+        """
+        Returns ``True`` if an encoder is found for the given type string
+        ``type_str``.  Otherwise, returns ``False``.  Raises
+        :class:`~eth_abi.exceptions.MultipleEntriesFound` if multiple encoders
```

Comment:
Right, I think there is a distinction that is worth being really clear about in the exception docstrings:
- `NoEntriesFound` is *probably* a bug with the string type you are sending to eth-abi, unless you just forgot to add a type to the registry, but that should be apparent to the caller
- `MultipleEntriesFound` is a bug with a custom registry you built (can it be anything else?)

Reply:
Yeah, expanding docstrings for those might save us some support effort in github issues down the road.  `MultipleEntriesFound`

https://github.com/ethereum/eth-abi/blob/169600c01b3f991c4d2cc739d1b9492c254503a2/eth_abi/registry.py#L93-L97

Expanding that to include the codec values as well might help, as well as a more verbose error messages:

> Multiple codecs were found for the type string "something".  The matching codecs and predicates which matched were:  (<CodecClassA>, <function: is_codec_class_A>), (...).  This error happens when there are two codec classes registered for the same type string.  You may need to unregister one of these codecs or create a custom matching function for your codec.  See the documentation on 'registering custom codecs' for more information.

Reply:
@carver @pipermerriam Yeah, that's the distinction, and I think the reason for this entire pull request essentially.  One exception is probably okay and easily recovered from.  The other indicates that the user of the library just did something wrong.  It's an expected property of a registry that the different registrations cover completely distinct regions of the input domain.

Reply:
@pipermerriam At the point where that error is raised, there's no information available about what value is being encoded or decoded.  I'm also not so sure that information would really be very helpful since the registration matching happens completely independently of the particular value being coded.

Reply:
> @pipermerriam At the point where that error is raised, there's no information available about what value is being encoded or decoded. I'm also not so sure that information would really be very helpful since the registration matching happens completely independently of the particular value being coded.

Shouldn't the type string be available? I agree that the type string is the only thing relevant for either of these exceptions.

Reply:
@carver The type string is already included in the error message.

Reply:
I don't see where he was suggesting that the values go into the message. Anyway, doesn't matter, LGTM in current state :+1:

Reply:
@carver Okay, actually I think he meant the *coders* associated with the registrations.  There's some confusion about terminology here since the registry tracks coders (i.e. encoders or decoders) and not codecs.

Reply:
Yeah, I was thinking about including the actual coder classes.  Current message looks fine.